drivers/platform/surface/surface_platform_profile.c

```diff
@@ -105,6 +130,28 @@ static int convert_profile_to_ssam(struct ssam_device *sdev, enum platform_profi
 	}
 }
 
+static int convert_profile_to_ssam_fan(struct ssam_device *sdev, enum platform_profile_option p)
```

Comment:
An alternative to this function would be to just use the `ssam_tmp_profile` to index into a lookup table to convert to the fan profile. But I went this route in case we end up wanting to expose decoupled fan profile / platform profile support.